packages/ifpack2/test/belos/build_problem.hpp

```diff
@@ -222,6 +232,61 @@ build_problem (Teuchos::ParameterList& test_params,
     x->putScalar (STS::zero ());
   }
 
+  if (useZoltan2) {
+#if defined(HAVE_IFPACK2_XPETRA) && defined(HAVE_IFPACK2_ZOLTAN2)
+    // Create an input adapter for the Tpetra matrix.
+    Zoltan2::XpetraCrsMatrixAdapter<crs_matrix_type>
+      zoltan_matrix(A);
+
+    // Specify partitioning parameters
+    Teuchos::ParameterList zoltan_params;
+    zoltan_params.set("partitioning_approach", "partition");
+    //
+    if (useParMETIS) {
+      if (comm->getRank() == 0) {
+        std::cout << "Using Zoltan2(ParMETIS)" << std::endl;
+      }
+      zoltan_params.set("algorithm", "parmetis");
+      zoltan_params.set("symmetrize_input", "transpose");
+      zoltan_params.set("partitioning_objective", "minimize_cut_edge_weight");
+    } else {
+      if (comm->getRank() == 0) {
+        std::cout << "Using Zoltan2(HyperGraph)" << std::endl;
+      }
+      zoltan_params.set("algorithm", "phg");
+    }
+
+    // Create and solve partitioning problem
+    Zoltan2::PartitioningProblem<Zoltan2::XpetraCrsMatrixAdapter<crs_matrix_type>>
+      problem(&zoltan_matrix, &zoltan_params);
+    problem.solve();
+
+    // Redistribute matrix
+    RCP<crs_matrix_type> zoltan_A;
+    zoltan_matrix.applyPartitioningSolution (*A, zoltan_A, problem.getSolution());
+    // Set it as coefficient matrix
+    A = zoltan_A;
+
+    // Redistribute RHS
+    RCP<TMV> zoltan_b;
+    Zoltan2::XpetraMultiVectorAdapter<TMV> adapterRHS(rcpFromRef (*b));
+    adapterRHS.applyPartitioningSolution (*b, zoltan_b, problem.getSolution());
+    // Set it as RHS
+    b = zoltan_b;
+
+    // Redistribute Sol
+    RCP<TMV> zoltan_x;
+    Zoltan2::XpetraMultiVectorAdapter<TMV> adapterSol(rcpFromRef (*x));
+    adapterSol.applyPartitioningSolution (*x, zoltan_x, problem.getSolution());
+    // Set it as Sol
+    x = zoltan_x;
+#else
+    TEUCHOS_TEST_FOR_EXCEPTION(
+      useZoltan2, std::invalid_argument,
+      "Both Xpetra and Zoltan2 are neeeded to usee Zoltan2.");
```

Comment:
```suggestion
      "Both Xpetra and Zoltan2 are needed to use Zoltan2.");
```

Reply:
Not sure:
https://github.com/trilinos/Trilinos/blob/401873f5da440ebe7a6e8c4569f3dd6e2fbec2c6/packages/ifpack2/cmake/Dependencies.cmake#L2
and
https://github.com/trilinos/Trilinos/blob/401873f5da440ebe7a6e8c4569f3dd6e2fbec2c6/packages/ifpack2/cmake/Ifpack2_config.h.in#L94-L98

Reply:
Thank you. I could use Zoltan2 without having it in the list. But I locally made the fixes, so I'll push once the auto-tester finish.